charts/pact-broker/templates/deployment.yaml

```diff
@@ -93,10 +93,16 @@ spec:
             - name: PACT_BROKER_DATABASE_USERNAME
               value: {{ include "broker.databaseUser" . }}
             - name: PACT_BROKER_DATABASE_PASSWORD
+              {{- if and .Values.postgresql.enabled .Values.postgresql.auth.password }}
+              value: {{ .Values.postgresql.auth.password | quote }}
+              {{- else if and .Values.externalDatabase.enabled .Values.externalDatabase.config.auth.password }}
```

Comment:
the `and` operator in this if statement prevents [cloudsql postgres iam passwordless authentication](https://cloud.google.com/sql/docs/postgres/authentication#authentication_for) from working -- as opposed to something like 
```yaml
{{- ternary .Values.postgresql.auth.password .Values.externalDatabase.config.auth.password .Values.postgresql.enabled | quote -}}
```

ex: using these values
```
externalDatabase:
  enabled: true
  config:
    host: "127.0.0.1"
    port: "5432"
    adapter: postgres
    databaseName: pact

    auth:
      username: "pact-broker-dev@dev-gcp-project.iam"
      password: "" # passwordless cloudsql iam auth
```

results in an invalid secret object. using a random placeholder string results in this error
```
pact-broker-67966bbd86-5b4cj pact-broker ! Unable to load application: Sequel::DatabaseConnectionError: PG::ConnectionBad: connection to server at "127.0.0.1", port 5432 failed: Connection refused
```

Reply:
@joshuasimon-taulia  How does passwordless authentication work with the Pact Broker anyways? It has to have a username and password as far as I know

Reply:
if you use these values or something similar to run a [cloudsql-proxy](https://github.com/GoogleCloudPlatform/cloud-sql-proxy) sidecar
```
extraContainers:
  - name: cloudsql-proxy
    image: gcr.io/cloudsql-docker/gce-proxy:1.22.0
    command:
      - /cloud_sql_proxy
      - '-instances=my-project:my-zone:my-db-name=tcp:5432'
      - '-enable_iam_login'
```
cloudsql-proxy authenticates to the db using the GCP IAM credentials bound to the pact broker pod's service account and exposes postgres on port 5432 or optionally as a socket on a shared volume mount. connecting with `PACT_BROKER_DATABASE_PASSWORD=""` was then working with the changeset in https://github.com/pact-foundation/pact-broker-chart/pull/44/files#diff-9b40c5793a6365436c02520da0ecf44ee7712bc048ee54a4d117831dc0a855d8R96-R98. to be fair, i figured out a workaround where i set `password` to an empty space to satisfy the if condition in the helm logic
```
externalDatabase:
    auth:
      password: " " # passwordless cloudsql iam auth
```

Reply:
So I've tried your changes in https://github.com/pact-foundation/pact-broker-chart/pull/44 locally, and when I enable `postgres` and also provide no password. I expect the Chart to generate a password and then use that in the deployment. But the password field is blank. This is the tricky situation where the subchart for postgres will generate the password for you and then use it in the deployment.

Reply:
```yaml
---
# Source: pact-broker/charts/postgresql/templates/secrets.yaml
apiVersion: v1
kind: Secret
metadata:
  name: release-name-postgresql
  namespace: "default"
  labels:
    app.kubernetes.io/name: postgresql
    helm.sh/chart: postgresql-11.9.13
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
type: Opaque
data:
  postgres-password: "SUJqWWpzVHp0Uw=="
  password: "Rk9xd1VjYkpzVQ=="
  # We don't auto-generate LDAP password when it's not provided as we do for other passwords
```

```yaml
            - name: PACT_BROKER_DATABASE_PASSWORD
              value: ""
```

Reply:
ah ok. ill just continue using my workaround